lib/Base/Type.cpp

```diff
@@ -54,4 +54,43 @@ llvm::raw_ostream &operator<<(llvm::raw_ostream &os, const TypeRef &type) {
   }
   return os << *type;
 }
+
+void Type::dump(llvm::raw_ostream &out) const {
+  out << "Type structure :\n";
+  out << this << "\n";
+}
+
+void Type::dump() const { dump(llvm::outs()); }
+
+std::string Type::toString() const {
```

Comment:
You should probably use `llvm::raw_string_ostream` here. Then you can just do 
```c++
std::string storage;
llvm::raw_string_ostream os(storage);
os << this; 
return os.str();
```

This way you don't need to duplicate the printing logic.